src/biom.cpp

```diff
@@ -77,30 +119,81 @@ biom::biom(std::string filename) {
                 sizeof(unsigned int) * n_obs, __FILE__, __LINE__);
         exit(EXIT_FAILURE);
     }
+}
 
-    uint32_t *current_indices = NULL;
-    double *current_data = NULL;
-    for(unsigned int i = 0; i < obs_ids.size(); i++)  {
-        std::string id_ = obs_ids[i];
-        unsigned int n = get_obs_data_direct(id_, current_indices, current_data);
-        obs_counts_resident[i] = n;
-        obs_indices_resident[i] = current_indices;
-        obs_data_resident[i] = current_data;
-    }
-    sample_counts = get_sample_counts();
+biom::biom() : has_hdf5_backing(false) { 
+    n_obs = 0;
+    malloc_resident(0);
 }
 
-biom::~biom() {
-    for(unsigned int i = 0; i < n_obs; i++) {
-        free(obs_indices_resident[i]);
-        free(obs_data_resident[i]);
+// not using const on indices/indptr/data as the pointers are being borrowed
```

Comment:
...we probably could still use const but i think it would require updating the object definition to change the member variables themselves to be const